sync3/extensions/receipts.go

```diff
@@ -95,18 +98,29 @@ func (r *ReceiptsRequest) ProcessInitial(ctx context.Context, res *Response, ext
 			internal.GetSentryHubFromContextOrDefault(ctx).CaptureException(err)
 			continue
 		}
-		// always include your own receipts
-		ownReceipts, err := extCtx.Store.ReceiptTable.SelectReceiptsForUser(roomID, extCtx.UserID)
-		if err != nil {
-			logger.Err(err).Str("user", extCtx.UserID).Str("room", roomID).Msg("failed to SelectReceiptsForUser")
-			internal.GetSentryHubFromContextOrDefault(ctx).CaptureException(err)
-			continue
-		}
-		if len(receipts) == 0 && len(ownReceipts) == 0 {
+		otherReceipts[roomID] = receipts
+		interestedRoomIDs = append(interestedRoomIDs, roomID)
+	}
+	// single shot query to pull out our own receipts for these rooms to always include our own receipts
+	ownReceipts, err := extCtx.Store.ReceiptTable.SelectReceiptsForUser(interestedRoomIDs, extCtx.UserID)
+	if err != nil {
+		logger.Err(err).Str("user", extCtx.UserID).Strs("rooms", interestedRoomIDs).Msg("failed to SelectReceiptsForUser")
+		internal.GetSentryHubFromContextOrDefault(ctx).CaptureException(err)
+		return
+	}
+
+	// move all own receipts into other receipts so we don't need to handle cases where receipts are in one map but not the other
+	for roomID, ownRecs := range ownReceipts {
+		otherReceipts[roomID] = append(otherReceipts[roomID], ownRecs...)
+	}
+
+	for roomID, receipts := range otherReceipts {
+		if len(receipts) == 0 {
 			continue
 		}
-		rooms[roomID], _ = state.PackReceiptsIntoEDU(append(receipts, ownReceipts...))
+		rooms[roomID], _ = state.PackReceiptsIntoEDU(append(receipts, receipts...))
```

Comment:
This doesn't look right. Should this be `state.PackReceiptsIntoEDU(receipts)`?

Reply:
Yes, I forgot this bit. This would've duplicated receipts which is why tests were still happy.

Reply:
```suggestion
		rooms[roomID], _ = state.PackReceiptsIntoEDU(receipts)
```